src/auth/google.strategy.ts

```diff
@@ -0,0 +1,48 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { PassportStrategy } from '@nestjs/passport';
+import { Strategy, VerifyCallback } from 'passport-google-oauth20';
+import googleOauthConfig from './google-oauth.config';
+import { ConfigType } from '@nestjs/config';
+import { AuthService } from './auth.service';
+
+@Injectable()
+export class GoogleStrategy extends PassportStrategy(Strategy) {
+  constructor(
+    @Inject('CONFIGURATION(googleOAuth)')
+    private readonly googleOAuthConfig: {
+        clientID: string;
+        clientSecret: string;
+        callbackURL: string;
+      },
+    private authService: AuthService,
+  ) {
+    super({
+      clientID: googleOAuthConfig.clientID,
+      clientSecret: googleOAuthConfig.clientSecret,
+      callbackURL: googleOAuthConfig.callbackURL,
+      scope: ['email', 'profile'],
+    });
+  }
+
+  async validate(
+    accessToken: string,
+    refreshToken: string,
+    profile: any,
+    done: VerifyCallback,
+  ) {
+    console.log({ profile });
+    const user = await this.authService.validateGoogleUser({
+      id: profile.id,
+      email: profile.emails[0].value,
+      firstName: profile.name.givenName,
+      lastName: profile.name.familyName,
+      storeName: "",
```

Comment:
Please add comment `// TODO: change tel` or something to note for yourself in the future